src/shogun/mathematics/graph/ops/shogun/details/add_avx.cpp

```diff
@@ -4,43 +4,82 @@
  * Authors: Gil Hoben
  */
 
-#include <unsupported/Eigen/CXX11/Tensor>
+#include <algorithm>
+#include <functional>
+#include <Eigen/Core>
+#include "PacketType.h"
+
+using namespace Eigen::internal;
 
 namespace shogun::graph::op {
+
+	struct Packet
+	{
+		aligned_vector m_data;
+		const size_t m_size;
+	};
+
 	template <typename T>
 	void add_kernel_implementation_avx(
-	    void* input1, void* input2, void* output, const size_t size);
+	    void* input1, void* input2, void* output);
 
 	template <typename T>
 	void add_kernel_implementation_avx(
-	    void* input1, void* input2, void* output, const size_t size)
+	    void* input1, void* input2, void* output)
 	{
-		Eigen::TensorMap<Eigen::Tensor<T, 1>> A(static_cast<T*>(input1), size);
-		Eigen::TensorMap<Eigen::Tensor<T, 1>> B(static_cast<T*>(input2), size);
-		Eigen::TensorMap<Eigen::Tensor<T, 1>> Out(static_cast<T*>(output), size);
+		using vector_type = typename alignedvector_from_builtintype<T, AVX_BYTESIZE>::type;
+		const auto& vec1 = std::get<vector_type>(static_cast<const Packet*>(input1)->m_data);
+		const auto& vec2 = std::get<vector_type>(static_cast<const Packet*>(input2)->m_data);
+		__m256i result;
 
-		Out.device(Eigen::DefaultDevice{}) = A + B;
+		std::transform(
+		    reinterpret_cast<const T*>(&vec1),
+		    reinterpret_cast<const T*>(&vec1) + AVX_BYTESIZE/sizeof(T),
+		    reinterpret_cast<const T*>(&vec2),
+		    reinterpret_cast<T*>(&result),
+		    std::plus<T>());
+
+		static_cast<Packet*>(output)->m_data = result;
 	}
```

Comment:
I have no idea how the compiler/runtime manage to get this to work... there are so many code smells here but this is a way to generalise integer math when there are no intrinsics functions